app/test/prepare_proposal_context_test.go

```diff
@@ -0,0 +1,102 @@
+package app_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/pkg/user"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+// TestTimeInPrepareProposalContext checks for an edge case where the block time
+// needs to be included in the sdk.Context that is being used in the
+// antehandlers. If a time is not included in the context, then the second
+// transaction in this test will always be filtered out, result in vesting
+// accounts never being able to spend funds.
+func TestTimeInPrepareProposalContext(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping DebugVestingAccountsTime test in short mode.")
```

Comment:
[question] is this name supposed to match the test name?

```suggestion
		t.Skip("skipping TestTimeInPrepareProposalContext test in short mode.")
```